packages/nest/src/schematics/library/library.ts

```diff
@@ -0,0 +1,140 @@
+import { normalize, Path } from '@angular-devkit/core';
+import {
+  apply,
+  chain,
+  externalSchematic,
+  filter,
+  MergeStrategy,
+  mergeWith,
+  move,
+  noop,
+  Rule,
+  SchematicContext,
+  template,
+  Tree,
+  url
+} from '@angular-devkit/schematics';
+import {
+  formatFiles,
+  getProjectConfig,
+  names,
+  offsetFromRoot,
+  toFileName,
+  updateJsonInTree,
+  updateWorkspaceInTree,
+  getNpmScope
+} from '@nrwl/workspace';
+import { Schema } from './schema';
+
+export interface NormalizedSchema extends Schema {
+  name: string;
+  prefix: string;
+  fileName: string;
+  projectRoot: Path;
+  projectDirectory: string;
+  parsedTags: string[];
+}
+
+export default function(schema: NormalizedSchema): Rule {
+  return (host: Tree, context: SchematicContext) => {
+    const options = normalizeOptions(host, schema);
+
+    return chain([
+      externalSchematic('@nrwl/workspace', 'lib', schema),
+      createFiles(options),
+      updateTsConfig(options),
+      addProject(options),
+      formatFiles(options)
+    ]);
+  };
+}
+
+function normalizeOptions(host: Tree, options: Schema): NormalizedSchema {
+  const defaultPrefix = getNpmScope(host);
+  const name = toFileName(options.name);
+  const projectDirectory = options.directory
+    ? `${toFileName(options.directory)}/${name}`
+    : name;
+
+  const projectName = projectDirectory.replace(new RegExp('/', 'g'), '-');
+  const fileName = projectName;
+  const projectRoot = normalize(`libs/${projectDirectory}`);
+
+  const parsedTags = options.tags
+    ? options.tags.split(',').map(s => s.trim())
+    : [];
+
+  const normalized: NormalizedSchema = {
+    ...options,
+    prefix: defaultPrefix, // we could also allow customizing this
+    fileName,
+    name: projectName,
+    projectRoot,
+    projectDirectory,
+    parsedTags
+  };
+
+  return normalized;
+}
+
+function createFiles(options: NormalizedSchema): Rule {
+  return mergeWith(
+    apply(url(`./files/lib`), [
+      template({
+        ...options,
+        ...names(options.name),
+        tmpl: '',
+        offsetFromRoot: offsetFromRoot(options.projectRoot)
+      }),
+      move(options.projectRoot),
+      options.unitTestRunner === 'none'
+        ? filter(file => !file.endsWith('spec.ts'))
+        : noop(),
+      options.publishable
+        ? noop()
+        : filter(file => !file.endsWith('package.json')),
+      options.service ? noop() : filter(file => !file.endsWith('service.ts')),
+      options.controller
+        ? noop()
+        : filter(file => !file.endsWith('controller.ts'))
+    ]),
+    MergeStrategy.Overwrite
+  );
+}
+
+function updateTsConfig(options: NormalizedSchema): Rule {
```

Comment:
Not needed when using the `@nrwl/node:lib` schematic.

Reply:
@Cammisuli if I switch to using @nrwl/node my two tests `not nested - should create a local tsconfig.json` and `nested - should create a local tsconfig.json` both fail because jest types are missing. That is why I put the updateTsConfig in there.

Reply:
Remove the `tsconfig.json` and any other file that's not needed from the templates. Those are included with the node:lib schematic. When you have your templates, they overwrite the ones that are already generated 😄

Reply:
Of course! That actually makes a lot of sense 🤦‍♂